src/clusterfuzz/_internal/cron/schedule_fuzz.py

```diff
@@ -0,0 +1,227 @@
+  assert quota['limit'], quota
+
+  return quota['limit'] - quota['usage']
+
+
+class BaseFuzzTaskScheduler:
+  """Base fuzz task scheduler for any deployment of ClusterFuzz."""
+
+  def __init__(self, num_cpus):
+    self.num_cpus = num_cpus
+
+  def get_fuzz_tasks(self):
+    raise NotImplementedError('Child class must implement.')
+
+  def _get_cpus_per_fuzz_job(self, job_name):
+    del job_name
+    # TODO(metzman): Actually implement this.
+    return 2
+
+
+class FuzzerJob:
+  """Data class that holds more info about FuzzerJobs than the ndb.Models do.
+  Something like this would probably not be needed if we were using SQL and
+  could use joins."""
+
+  def __init__(self, job, project, queue, fuzzer=None, weight=None):
+    self.job = job
+    self.project = project
+    self.queue = queue
+    self.fuzzer = fuzzer
+    self.weight = weight
+
+  def copy(self):
+    return FuzzerJob(
+        job=self.job,
+        project=self.project,
+        queue=self.queue,
+        fuzzer=self.fuzzer,
+        weight=self.weight)
+
+
+class OssfuzzFuzzTaskScheduler(BaseFuzzTaskScheduler):
+  """Fuzz task scheduler for OSS-Fuzz."""
+
+  def get_fuzz_tasks(self) -> Dict[str, tasks.Task]:
+    # TODO(metzman): Handle high end.
+    # A job's weight is determined by its own weight and the weight of the
+    # project is a part of. First get project weights.
+    logs.info('Getting projects.')
+    projects = list(
+        ndb_utils.get_all_from_query(data_types.OssFuzzProject.query()))
+
+    total_cpu_weight = sum(project.cpu_weight for project in projects)
+    project_weights = {}
+    for project in projects:
+      project_weight = project.cpu_weight / total_cpu_weight
+      project_weights[project.name] = project_weight
+
+    # Then get FuzzerJob weights.
+    logs.info('Getting jobs.')
+    jobs = {}
+    for job in ndb_utils.get_all_from_query(data_types.Job.query()):
+      jobs[job.name] = FuzzerJob(
+          job=job.name,
+          project=job.project,
+          queue=tasks.queue_for_platform(job.platform))
+
+    fuzzer_job_weight_by_project = collections.defaultdict(int)
+    fuzzer_jobs = {}
+    fuzzer_job_query = ndb_utils.get_all_from_query(
+        data_types.FuzzerJob.query())
+
+    def get_fuzzer_job_key(fuzzer, job):
+      return f'{job},{fuzzer}'
+
+    for fuzzer_job_db in fuzzer_job_query:
+      fuzzer_job = jobs[fuzzer_job_db.job].copy()
+      fuzzer_job.fuzzer = fuzzer_job_db.fuzzer
+      project_weight = project_weights.get(fuzzer_job.project, None)
+      if project_weight is None:
+        logs.info(f'No project weight for {fuzzer_job.project}')
+        continue
+
+      fuzzer_job.weight = fuzzer_job_db.actual_weight * project_weight
+      key = get_fuzzer_job_key(fuzzer_job_db.fuzzer, fuzzer_job_db.job)
+      fuzzer_jobs[key] = fuzzer_job
+
+      fuzzer_job_weight_by_project[fuzzer_job.project] += (
+          fuzzer_job_db.actual_weight)
+
+    for key, fuzzer_job in list(fuzzer_jobs.items()):
```

Comment:
Do we actually need this key? Or even this `fuzzer_jobs` dict at all? Can it just be a list instead?

Reply:
Youre right. I think there was some left over structure from my initial implementations which were more complex since they didn't have the FuzzerJob class.